### ISSUE TITLE
Samsung s4 / s5 blocks CC number input

### ISSUE THREAD

**Issue by clautour** [braintree/braintree-web#248]
Hi, 

We've encounded an issue during testing where a user is unable to complete entry of their credit card number (hosted fields) on samsung s4 / s5 devices.

The user is able to input the first four digits of the number `4111` the input field then autoformats to include a break before the next four digits - however at this point any character inputs (besides backspace) are ignored.

![image](https://cloud.githubusercontent.com/assets/12725790/22007692/c1080aa8-dcaf-11e6-88e1-be566e050b86.png)

**Comment by crookedneighbor** [comment]
What version of braintree-web are you using? The latest version is v3.6.3 and includes some fixes for Samsung keyboard issues like this one.

**Comment by clautour** [comment]
Sorry I forgot that critical bit of info - we are on v3.6.3.

**Comment by crookedneighbor** [comment]
Thanks for the info, we're looking into it.

Can you tell me what keyboard the device is using?

In the meantime, you can turn off formatting for Android devices like this:

```js
var isAndroid = navigator.userAgent.toLowerCase().indexOf('android') !== -1;

braintree.hostedFields.create({
  client: clientInstance,
  fields: {
    number: {
      selector: '#my-cc',
      formatInput: !isAndroid
    },
    // ...
  }
}, /* ... */);
```

**Comment by clautour** [comment]
Thanks for the work around.

The issue is observed using the default Samsung keyboard.

**Comment by clautour** [comment]
We've just had a customer report an issue with an S6 as well.

**Comment by skunkworks** [comment]
@clautour Can you verify that this behavior is occurring in the Samsung default web browser and not in Chrome?

**Comment by skunkworks** [comment]
We have made a [change to the restricted-input library](https://github.com/braintree/restricted-input/pull/31) that disables field formatting for the Samsung Internet browser. We will update the JS SDK with this fix shortly.

**Comment by clautour** [comment]
Thanks for the update. We observed the issue in multiple browsers: Samsung, Chrome, and our application which uses `android.webkit.WebView`.

Re the PR:
> Some (random?) numbers would often be dropped from the credit card number field 

This _could_ describe an issue we saw before upgrading to 3.6.3; where by the formatting would replace the  5th,9th,13th digit of the CC number with whitespace, rather than insert the whitespace before it.

**Comment by intelliot** [comment]
@clautour That sounds similar to an issue we fixed in 3.6.3. As mentioned earlier, an upcoming update will disable input formatting in Samsung browsers - we'll update here when this is available.

**Comment by skunkworks** [comment]
@clautour We have fixed this issue in version [3.7.0](https://github.com/braintree/braintree-web/releases/tag/3.7.0), which disables input formatting for the Samsung Internet browser. Please let us know if you continue having issues after upgrading. Thanks!

**Comment by clautour** [comment]
Thanks for the update, we'll update and let you know if the problem continues or is resolved.
